certora/ArrangerConduit.spec

```
@@ -0,0 +1,710 @@
+    env e;
+
+    address arrangerBefore = arranger();
+    address registryBefore = registry();
+    address rolesBefore = roles();
+
+    file(e, what, data);
+
+    address arrangerAfter = arranger();
+    address registryAfter = registry();
+    address rolesAfter = roles();
+
+    assert what == to_bytes32(0x617272616e676572000000000000000000000000000000000000000000000000)
+           => arrangerAfter == data, "file did not set arranger";
+    assert what != to_bytes32(0x617272616e676572000000000000000000000000000000000000000000000000)
+           => arrangerAfter == arrangerBefore, "file did not keep unchanged arranger";
+
+    assert what == to_bytes32(0x7265676973747279000000000000000000000000000000000000000000000000)
+           => registryAfter == data, "file did not set registry";
+    assert what != to_bytes32(0x7265676973747279000000000000000000000000000000000000000000000000)
+           => registryAfter == registryBefore, "file did not keep unchanged registry";
+
+    assert what == to_bytes32(0x726f6c6573000000000000000000000000000000000000000000000000000000)
+           => rolesAfter == data, "file did not set roles";
+    assert what != to_bytes32(0x726f6c6573000000000000000000000000000000000000000000000000000000)
+           => rolesAfter == rolesBefore, "file did not keep unchanged roles";
+}
+
+// Verify revert rules on file
+rule file_address_revert(bytes32 what, address data) {
+    env e;
+
+    mathint wardsSender = wards(e.msg.sender);
+
+    file@withrevert(e, what, data);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = wardsSender != 1;
+    bool revert3 = what != to_bytes32(0x617272616e676572000000000000000000000000000000000000000000000000)
+                && what != to_bytes32(0x7265676973747279000000000000000000000000000000000000000000000000)
+                && what != to_bytes32(0x726f6c6573000000000000000000000000000000000000000000000000000000);
+
+    assert lastReverted <=> revert1 || revert2 || revert3, "Revert rules failed";
+}
+
+// Verify correct storage changes for non reverting setBroker
+rule setBroker(address usr, address asset, bool valid) {
+    env e;
+
+    address otherUsr;
+    require otherUsr != usr;
+
+    address otherAsset;
+    require otherAsset != asset;
+    
+    bool isBrokerOtherUsrBefore = isBroker(otherUsr, asset);
+    bool isBrokerOtherAssetBefore = isBroker(usr, otherAsset);
+    
+    setBroker(e, usr, asset, valid);
+
+    bool isBrokerUsrAfter = isBroker(usr, asset);
+    bool isBrokerOtherUsrAfter = isBroker(otherUsr, asset);
+    bool isBrokerOtherAssetAfter = isBroker(usr, otherAsset);
+
+    assert isBrokerUsrAfter == valid, "setBroker did not set broker to valid";
+    assert isBrokerOtherUsrAfter == isBrokerOtherUsrBefore, "setBroker did not keep unchanged the rest of brokers for the same asset";
+    assert isBrokerOtherAssetAfter == isBrokerOtherAssetBefore, "setBroker did not keep unchanged the rest of brokers for other assets";
+}
```

I solve this cases using the following:

```suggestion
// Verify correct storage changes for non reverting setBroker
rule setBroker(address usr, address asset, bool valid) {
    env e;

    address otherUsr;
    address otherAsset;
    require otherUsr != usr || otherAsset != asset;
    
    bool isBrokerOtherBefore = isBroker(otherUsr, otherAsset);
    
    setBroker(e, usr, asset, valid);

    bool isBrokerUsrAfter = isBroker(usr, asset);
    bool isBrokerOtherAfter = isBroker(otherUsr, otherAsset);

    assert isBrokerUsrAfter == valid, "setBroker did not set brokers[usr][asset] to valid";
    assert isBrokerOtherAfter == isBrokerOtherBefore, "setBroker did not keep unchanged the rest of brokers[x][y]";
}
```
Not only because it is shorter, but also because the way done now it is excluding the case where `isBroker(otherUser, otherAsset)`.